app/models/outgoing_email.rb

```diff
@@ -36,6 +36,7 @@ class OutgoingEmail < ApplicationRecord
 
   belongs_to :client
   belongs_to :user, optional: true
+  has_many :bulk_client_message_outgoing_emails, dependent: :destroy
```

Comment:
the outgoing emails get deleted when a client gets deleted, and then if those are still present in the db on another table, it blocks deletion. This only deletes the entry that's FOR that email_id. It feels a little weird to do this, because we lose the record that that client was messaged (even though they're deleted), but since we use a reference to create the relationship i dont know if theres a way to retain it...